x/slashing/keeper_test.go

```diff
@@ -0,0 +1,111 @@
+	require.Equal(t, int64(0), info.StartHeight)
+	require.Equal(t, int64(0), info.IndexOffset)
+	require.Equal(t, int64(0), info.SignedBlocksCounter)
+	require.Equal(t, int64(0), info.JailedUntil)
+	height := int64(0)
+	// 1000 blocks OK
+	for ; height < 1000; height++ {
+		ctx = ctx.WithBlockHeight(height)
+		keeper.handleValidatorSignature(ctx, val, true)
+	}
+	info, found = keeper.getValidatorSigningInfo(ctx, val.Address())
+	require.True(t, found)
+	require.Equal(t, int64(0), info.StartHeight)
+	require.Equal(t, SignedBlocksWindow, info.SignedBlocksCounter)
+	// 50 blocks missed
+	for ; height < 1050; height++ {
+		ctx = ctx.WithBlockHeight(height)
+		keeper.handleValidatorSignature(ctx, val, false)
+	}
+	info, found = keeper.getValidatorSigningInfo(ctx, val.Address())
+	require.True(t, found)
+	require.Equal(t, int64(0), info.StartHeight)
+	require.Equal(t, SignedBlocksWindow-50, info.SignedBlocksCounter)
+	// validator should be bonded still
+	validator := sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Bonded, validator.GetStatus())
+	pool := sk.GetPool(ctx)
+	require.Equal(t, int64(100), pool.BondedTokens)
+	// 51st block missed
+	ctx = ctx.WithBlockHeight(height)
+	keeper.handleValidatorSignature(ctx, val, false)
+	info, found = keeper.getValidatorSigningInfo(ctx, val.Address())
+	require.True(t, found)
+	require.Equal(t, int64(0), info.StartHeight)
+	require.Equal(t, SignedBlocksWindow-51, info.SignedBlocksCounter)
+	// validator should have been revoked
+	validator = sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Unbonded, validator.GetStatus())
+	// unrevocation should fail prior to jail expiration
+	got = slh(ctx, NewMsgUnrevoke(addr))
+	require.False(t, got.IsOK())
+	// unrevocation should succeed after jail expiration
+	ctx = ctx.WithBlockHeader(abci.Header{Time: int64(86400 * 2)})
+	got = slh(ctx, NewMsgUnrevoke(addr))
+	require.True(t, got.IsOK())
+	// validator should be rebonded now
+	validator = sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Bonded, validator.GetStatus())
+	// validator should have been slashed
+	pool = sk.GetPool(ctx)
+	require.Equal(t, int64(99), pool.BondedTokens)
+	// validator start height should have been changed
+	info, found = keeper.getValidatorSigningInfo(ctx, val.Address())
+	require.True(t, found)
+	require.Equal(t, height, info.StartHeight)
+	require.Equal(t, SignedBlocksWindow-51, info.SignedBlocksCounter)
+	// validator should not be immediately revoked again
+	height++
+	ctx = ctx.WithBlockHeight(height)
+	keeper.handleValidatorSignature(ctx, val, false)
+	validator = sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Bonded, validator.GetStatus())
+	// validator should be revoked again after 100 unsigned blocks
+	nextHeight := height + 100
+	for ; height <= nextHeight; height++ {
+		ctx = ctx.WithBlockHeight(height)
+		keeper.handleValidatorSignature(ctx, val, false)
+	}
+	validator = sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Unbonded, validator.GetStatus())
```

Comment:
holy, let's use some whitespaces - I recommend an enter before each comment in this function, ex:
```golang
+	// validator start height should have been changed
+	info, found = keeper.getValidatorSigningInfo(ctx, val.Address())
+	require.True(t, found)
+	require.Equal(t, height, info.StartHeight)
+	require.Equal(t, SignedBlocksWindow-51, info.SignedBlocksCounter)

+	// validator should not be immediately revoked again
+	height++
+	ctx = ctx.WithBlockHeight(height)
+	keeper.handleValidatorSignature(ctx, val, false)
+	validator = sk.ValidatorByPubKey(ctx, val)
+	require.Equal(t, sdk.Bonded, validator.GetStatus())

+	// validator should be revoked again after 100 unsigned blocks
+	nextHeight := height + 100
+	for ; height <= nextHeight; height++ {
+		ctx = ctx.WithBlockHeight(height)
+		keeper.handleValidatorSignature(ctx, val, false)
+	}
```

Reply:
^ the content of this test looks good though

Reply:
Added whitespaces between comment-headed blocks in both test functions.